### ISSUE TITLE
solana [stake-account|vote-account] returns an error: Error: invalid type: null, expected struct EncodedConfirmedBlock

### ISSUE THREAD

**Issue by max-block** [solana-labs/solana#12843]
```
$ solana vote-account /path/to/vote.json
Error: invalid type: null, expected struct EncodedConfirmedBlock
```

solana stake-account returns the same error

solana --version
solana-cli 1.3.16 (src:b31ec057; feat:2415322184)

**Comment by CriesofCarrots** [comment]
Hi @max-block ; what cluster are you targeting? Are you willing to share the pubkey of `/path/to/vote.json`?

**Comment by max-block** [comment]
@CriesofCarrots testnet TDS, 95xoBAM5kHDANac3rim6ka1pUwrtVKFLdwzVLVTf8hZK

**Comment by max-block** [comment]
I've upgraded to v1.3.17, but the problem is still present.

**Comment by CriesofCarrots** [comment]
@max-block , you should find that `solana vote-account` works for you now.
@ericlwilliams , `block 40268260` is currently in limbo... it's been pruned from blockstore on the api nodes, but hasn't made it into bigtable from the warehouse node yet. However, the other block you were looking at, `block 40700256`, now works again.

**Comment by ericlwilliams** [comment]
Thanks @CriesofCarrots !

**Comment by CriesofCarrots** [comment]
Issue was caused by a bug in deserialization of legacy Reward data from rocksdb.
Consider this resolved: Long-term solution is at #12860 . Any remaining data gaps (like the block mentioned above) should be resolved as the testnet warehouse node writes these blocks/rewards into bigtable.
